internal/gmm/message/build.go

```diff
@@ -165,7 +165,11 @@ func BuildAuthenticationRequest(ue *context.AmfUe) ([]byte, error) {
 
 	m.GmmMessage.AuthenticationRequest = authenticationRequest
 
-	return m.PlainNasEncode()
+	m.SecurityHeader = nas.SecurityHeader{
+		ProtocolDiscriminator: nasMessage.Epd5GSMobilityManagementMessage,
+		SecurityHeaderType:    nas.SecurityHeaderTypeIntegrityProtectedAndCiphered,
+	}
+	return nas_security.Encode(ue, m, accessType)
```

Comment:
Will this make UE get a plain-nas message while nas sec header is SecurityHeaderTypeIntegrityProtectedAndCiphered when secCtx is not available?

Reply:
No, when security context is not exist, `nas_security.Encode()` use SpareHalfOctetAndSecurityHeaderType as security header type  and `m.SecurityHeader` is not used.

Reply:
Get it, thanks.